### PR TITLE
Big auth refactor

### PR DESCRIPTION
**Pull request by mnlbox**
- [x] add new DDL and docs
- [x] change `email` to `username` (we don't need #19 now)
- [x] add new environment variable `USER_MANAGEMENT_DATABASE_SCHEMA_NAME` and support schema name
- [x] refactor `/register`
- [x] rename `/sign-in` to `/login` and refactor
- [x] refactor `/refetch-token`
- [x] refactor and small fix on `/activate-account`
- [x] refactor `/new-password`
- [x] some code cleanup and remove unused variables
- [x] Prepare for multi role support
- [x] Move default role to database rather static `user` (now each user can have own default role)

### CODE REVIEW

File: src/auth/auth.js [hunk 1]
```diff
@@ -254,11 +257,11 @@ router.post('/new-password', async (req, res, next) => {
   res.send('OK');
 });
 
-router.post('/sign-in', async (req, res, next) => {
+router.post('/login', async (req, res, next) => {
```
Comment (elitan) [inline src/auth/auth.js]: Why the change from `sign-in` to `login`?

File: README.md [hunk 2]
```diff
@@ -12,22 +12,115 @@
...
+
+-- Table: user_management."user"
+
+-- DROP TABLE user_management."user";
+
+CREATE TABLE user_management."user"
+(
+    id uuid NOT NULL DEFAULT gen_random_uuid(),
+    username text COLLATE pg_catalog."default" NOT NULL,
+    password text COLLATE pg_catalog."default" NOT NULL,
```
Comment (elitan) [inline README.md]: why not stick with `password_hash`? It is more accurate.

File: README.md [hunk 3]
```diff
@@ -12,22 +12,115 @@
...
-  active boolean not null default false
-);
-
-CREATE TABLE IF NOT EXISTS refetch_tokens (
-  refetch_token uuid primary key,
-  user_id integer not null,
-  added_at timestamp with time zone DEFAULT now(),
-  FOREIGN KEY (user_id) REFERENCES users (id)
-);
+-- SCHEMA: user_management
```
Comment (elitan) [inline README.md]: In the readme I think we should make it as simple as possible. I think we should not introduce a new schema. That is for advanced users and we can describe how to use a different schema than `public` when we write about the newly introduced `USER_MANAGEMENT_DATABASE_SCHEMA_NAME` environment variable.

File: src/auth/auth.js [hunk 4]
```diff
@@ -245,7 +250,7 @@ router.post('/new-password', async (req, res, next) => {
     var hasura_data = await graphql_client.request(query, {
       username,
       password_hash,
```
Comment (elitan) [inline src/auth/auth.js]: In other places I think you have replaced this column with just `password`. Here it is `password_hash`. Has to be consistent.

**Comment by elitan** [comment]
In general I think this PR should be:

- Change email to username
- Introduce the possibility to use a different schema than the default (public)

and not change anything else

**Comment by mnlbox** [comment]
> In general I think this PR should be:
> 
>     * Change email to username
> 
>     * Introduce the possibility to use a different schema than the default (public)
> 
> 
> and not change anything else

Also we have some small bug that fixed here. For example activate not working and user with `activate = false` can login and some other small things.

File: src/auth/auth.js (hunk 1 above)
Comment (mnlbox) [reply src/auth/auth.js]: I want to uniform endpoint names .If we want use `sign-in` rather than `login` it's better to use `sign-up` rather than `register`.
I think:
`sign-up -> sign-in` (is beautiful)
`register -> login` (is beautiful)
but `register -> sign-in` (is a little messy for me :wink:)

Comment (elitan) [reply src/auth/auth.js]: Good point!

File: README.md (hunk 2 above)
Comment (elitan) [reply README.md]: Let's go with `password`. Seem to be the standard after looking at Wordpress and ghost schemas.

File: README.md (hunk 3 above)
Comment (mnlbox) [reply README.md]: You are right. Done it in new commit. :wink:

File: README.md [hunk 5]
```diff
@@ -12,21 +12,36 @@
...
-  email text not null UNIQUE,
-  password_hash text not null,
-  role text not null default 'user',
-  email_token uuid not null,
-  active boolean not null default false
+CREATE TABLE IF NOT EXISTS "user" (
+    id uuid NOT NULL DEFAULT gen_random_uuid() PRIMARY KEY,
+    username text NOT NULL UNIQUE,
+    password text COLLATE pg_catalog."default" NOT NULL,
+    active boolean NOT NULL,
```
Comment (elitan) [inline README.md]: should default to `false`

File: README.md [hunk 6]
```diff
@@ -12,21 +12,36 @@
...
-  added_at timestamp with time zone DEFAULT now(),
-  email text not null UNIQUE,
-  password_hash text not null,
-  role text not null default 'user',
-  email_token uuid not null,
-  active boolean not null default false
+CREATE TABLE IF NOT EXISTS "user" (
+    id uuid NOT NULL DEFAULT gen_random_uuid() PRIMARY KEY,
+    username text NOT NULL UNIQUE,
+    password text COLLATE pg_catalog."default" NOT NULL,
```
Comment (elitan) [inline README.md]: what is `COLLATE pg_catalog."default"`?

**Comment by elitan** [comment]
`activation_token` (former `email_token`) can be used for two things.

1. Activate a user.
2. Change password for a user.

We should come up with a better name for this variable.

File: README.md (hunk 6 above)
Comment (mnlbox) [reply README.md]: Oh I missed it. fixed :wink:

File: README.md (hunk 5 above)
Comment (mnlbox) [reply README.md]: done

**Comment by elitan** [comment]
Change name of `activation_token` to `secret_token`

I will take a look on all changes here soon.

**Comment by mnlbox** [comment]
I think it's ready to merge. Please take a look when you have time.
Thanks for your attention brow :wink:
